extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -0,0 +1,156 @@
+      JSONObject jsonObject = (JSONObject) jsonParser.parse( fr );
+      String jsonString = jsonObject.toJSONString();
+      fr.close();
+      return (JSONObject) jsonParser.parse( jsonString );
+    } catch ( ParseException e ) {
+      logger.error( "File: " + file.getAbsolutePath() + " is not valid json, and will not be processed" );
+      fr.close();
+      //rename the file here and allow the chronos container to die because of bad input
+      renameFile( file.getAbsolutePath() );
+    }
+    return null;
+  }
+
+  private File[] listFiles( final File folder, final String fileExtension ) throws IOException {
+    if ( folder.isDirectory() && folder.canRead() ) {
+      return folder.listFiles( new FileFilter() {
+        @Override
+        public boolean accept( File f ) {
+          return f.isFile() && f.getName().toLowerCase().endsWith( fileExtension );
+        }
+      } );
+    } else {
+      logger.error( folder.getAbsolutePath() + " is not a valid directory" );
+    }
+    return null;
+  }
+
+  private void renameFile( final String filename ) {
+    File dotJson = new File( filename );
+    File dotJsonDotCurTime = new File( filename + "." + System.currentTimeMillis() );
+    dotJson.renameTo( dotJsonDotCurTime );
+  }
+
+  /**
+   * System listener looks for JSON files at start up and runs them as work items
+   * @param session
+   * @return
+   */
+  @Override
+  public boolean startup( IPentahoSession session ) {
+    Response status;
+    Map<String, JSONObject> jsonFiles = new HashMap<>(); //map between the json objects and the filenames from which they were consumed
+    try {
+      if ( environmentVariablesFilepath != null ) {
+        JSONObject environmentVariables = getJsonFromFile( new File( environmentVariablesFilepath ) );
+        jsonFiles.put( environmentVariablesFilepath, environmentVariables );
+      } else {
+        jsonFiles = parseJsonFiles();
+      }
+      for ( String filename : jsonFiles.keySet() ) {
+        JSONObject json = jsonFiles.get( filename );
+        WorkItemRequestPayload payload = new WorkItemRequestPayload( json );
+        if ( !payload.isValid() ) {
+          logger.error( "File: " + filename + " is missing required parameters, skipping" );
+        } else {
+          status = issueRequest( payload );
+          if ( status == null ) {
+            logger.error( "File: " + filename + " has parameters formatted in an unparsable way. Could not process work item." );
+          }
+          //todo- sometype of status handling once WI lifecycle is defined
+        }
+        renameFile( filename );
+      }
+    } catch ( ParseException | IOException | SecurityException e ) {
+      logger.error( e.getMessage(), e );
+    }
+    return true;
+  }
+
+  private Response issueRequest( final WorkItemRequestPayload payload ) throws IOException, ParseException {
+    return new ActionResource().invokeAction(
+        ActionUtil.INVOKER_SYNC_VALUE,
```

Comment:
Just a thought: 

don't think we need an extra constant in ActionUtil for this. If we don't wan't to stick a magic number here ( which is the correct approach ), we can either

a)  pass a String.valueOf( true )
b)  declare it as a final static string within this systemListener class